code/modules/mob/living/carbon/human/life.dm

```diff
@@ -485,6 +485,8 @@
 	if(on_fire)
 		var/thermal_protection = get_thermal_protection()
 
+		if(wear_suit && wear_suit.flags & AUTOEXTINGUISH)
+			wear_suit.auto_extinguish(src)

```

Comment:
This is here because I wrote suit autoextinguishing as a feature that could potentially be written to be used by anybody, not just Plasmamen.